app/controllers/user_controller.php

```diff
@@ -73,4 +84,98 @@ public function logout()
         session_destroy();
         redirect('user/login');
     }
+
+    public function profile()
+    {
+        $user = new User();
+        $user->id = Param::get('user_id');
+        $user->getProfile();
+
+        if(!isset($user->username)) {
+            redirect('notfound/pagenotfound');
+        }
+
+        $threads_followed = array();
+        $thread_followed_id = Follow::getFollowedThreadIds($user->id);
+
+        foreach($thread_followed_id as $thread) {
+            $threads_followed[] = Thread::getById($thread['thread_id']);
+        }
```

Comment:
L99-103. To me, this seems like a method in User model as it is related to the threads being followed by the user.
Should be something like $user->getThreadsFollowed()
L106, should be a method in Thread and be named like getUser()
L109-113 should also be a method in User ie. $user->getThreadsCreated()